### ISSUE TITLE
Exceed maximum number of requests - Valves displayed but not working

### ISSUE THREAD

**Issue by fgomecs** [martijndierckx/homebridge-hydrawise#9]
Hi,
I just installed this plugin on a fresh raspian installed with the latest homebridge and notes. When I launch homebridge I see the valves showing up in homekit and homebridge UX but I get the following logs over and over. The logs are not a problem, the problem is that the valves don't work:

Unhandled rejection StatusCodeError: 429 - "Exceeded maximum number of requests. You cannot make more than 30 requests in any 5 minute period. You have made 199 calls in this period."
    at new StatusCodeError (/usr/lib/node_modules/homebridge-hydrawise/node_modules/request-promise/lib/errors.js:32:15)
    at Request.RP$callback [as _callback] (/usr/lib/node_modules/homebridge-hydrawise/node_modules/request-promise/lib/rp.js:77:29)
    at Request.self.callback (/usr/lib/node_modules/homebridge-hydrawise/node_modules/request/request.js:185:22)
    at Request.emit (events.js:310:20)
    at Request.<anonymous> (/usr/lib/node_modules/homebridge-hydrawise/node_modules/request/request.js:1154:10)
    at Request.emit (events.js:310:20)
    at IncomingMessage.<anonymous> (/usr/lib/node_modules/homebridge-hydrawise/node_modules/request/request.js:1076:12)
    at Object.onceWrapper (events.js:416:28)
    at IncomingMessage.emit (events.js:322:22)
    at endReadableNT (_stream_readable.js:1187:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
Unhandled rejection StatusCodeError: 429 - "Exceeded maximum number of requests. You cannot make more than 30 requests in any 5 minute period. You have made 200 calls in this period."
    at new StatusCodeError (/usr/lib/node_modules/homebridge-hydrawise/node_modules/request-promise/lib/errors.js:32:15)
    at Request.RP$callback [as _callback] (/usr/lib/node_modules/homebridge-hydrawise/node_modules/request-promise/lib/rp.js:77:29)
    at Request.self.callback (/usr/lib/node_modules/homebridge-hydrawise/node_modules/request/request.js:185:22)
    at Request.emit (events.js:310:20)
    at Request.<anonymous> (/usr/lib/node_modules/homebridge-hydrawise/node_modules/request/request.js:1154:10)
    at Request.emit (events.js:310:20)
    at IncomingMessage.<anonymous> (/usr/lib/node_modules/homebridge-hydrawise/node_modules/request/request.js:1076:12)
    at Object.onceWrapper (events.js:416:28)
    at IncomingMessage.emit (events.js:322:22)
    at endReadableNT (_stream_readable.js:1187:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)

**Comment by fgomecs** [comment]
Hello everyone,
I did a couple of tweaks in my Hydrawise account and now it works. This is what I did:

1) Removed all zones
2) Defined "standard configuration" instead of "advanced configuration"
3) Removed all "smart" and "weather" triggers to either soak, water more, avoid when wind, etc
4) Created one zone from scratched and defined watering time = 30 mins
4) Went back to homebridge, removed plugin, config, reset homebridge
5) Added plugin again, added config and changed "default_runtime": "1800", "polling_interval": "60000"
6) I made sure the default run time in homebridge matches what I put in Hydrawise

**Comment by fgomecs** [comment]
I am going to start adding parameters and zones until it fails again and let you know what is exactly causing the problem..

**Comment by fgomecs** [comment]
Ok, I just did my first edit, changed the name of the zone to "Left of Driveway" and noticed Homebridge only captures "Left of Drivewa" (missing the Y) so there could be a character limit. Now, this didn't stopped the sprinkler from working... Testing continues....

**Comment by fgomecs** [comment]
First failed test
1) Change from Standard to Advanced configuration mode for your controller
2) Change from Timebased to Smart ET

![ET](https://user-images.githubusercontent.com/25732501/80309168-3a8c9e80-87a1-11ea-9df9-c658c3a5dc04.PNG)
![ET 2](https://user-images.githubusercontent.com/25732501/80309170-3c566200-87a1-11ea-9553-f264d9316c4a.PNG)
![error](https://user-images.githubusercontent.com/25732501/80309199-7a538600-87a1-11ea-87d9-f05037aa25e2.PNG)

**Comment by fgomecs** [comment]
I switched from Smart ET to Virtual Solar Sync and it works. Seems Smart ET Scheduled is the troublemaker

![virtual 1](https://user-images.githubusercontent.com/25732501/80309273-09609e00-87a2-11ea-8f9e-9d3f327a465d.PNG)
![virtual 2](https://user-images.githubusercontent.com/25732501/80309275-0b2a6180-87a2-11ea-90c2-2c6a998c9f03.PNG)

**Comment by fgomecs** [comment]
Ok, I did all kinds of Advanced Configuration using Time-based schedules. I added weather station triggers, winds, rainfall, evaporation, I added schedules and start times and it continues to work. The moment I switch to Smart ET it fails. I will stop my testing here and I hope this helps others who struggled like I did. Thanks @martijndierckx for creating this amazing module. You are the man!

Stay safe during COVID19!

**Comment by fgomecs** [comment]
The last update, I found that the name of the zone DOES make this module fail...

I added all my 7 zones, used names with "space" (i.e. Flower Bed) and I experienced the same thing that started this issue. So I changed all my zones to one-word names (i.e. Flowers) and worked again.

Cheers!

**Comment by fgomecs** [comment]
latest update... unfortunately it keeps failing even again with 1 zone, no spaces, no fancy configs in the controller. Seems there is again an excessive amount of requests that overwhelms the controller. It works the first time, then I stopped the zone, and then I tried starting it again and it started failing....

I will let the experts figure this one out

**Comment by martijndierckx** [comment]
I've put a lot of effort into a new version of the hydrawise-api and homebridge plugin, allowing local communication to the controller and optimising the cloud connection as well.
Can you have a look if this new version works for you? I've tested it with a single controller primarily on a local connection, which works like a charm.

**Comment by fgomecs** [comment]
Sure! I’ll give it a try this week. Thanks man

**Comment by fgomecs** [comment]
You are my hero! Wow.... this is working perfectly now. I just updated, reconfigured using CLOUD added all of my 7 zones with Smart ET, Time-based, etc... All kinds of configurations in Hydrawize and no problems. The button ON/OFF toggles in Homekit works super quickly and accurately. Not only that, while I was doing changes in hydrawise the changes were being reflected in homekit almost real time! Before I had to restart homebridge. YOU ARE DA MAN!!!! I have a lot of respect for you. Just this weekend I was able to turn on one LED on my raspberry pi using homekit. I used code from other people and lots of reading... And here you are, doing all this from scratch. NICE! Thanks and awesome job man.
